src/stream/src/executor/source/state_table_handler.rs

```diff
@@ -180,11 +185,7 @@ impl<S: StateStore> SourceStateTableHandler<S> {
         Ok(())
     }
 
-    /// This function provides the ability to persist the source state
-    /// and needs to be invoked by the ``SourceReader`` to call it,
-    /// and will return the error when the dependent ``StateStore`` handles the error.
-    /// The caller should ensure that the passed parameters are not empty.
-    pub async fn take_snapshot<SS>(&mut self, states: Vec<SS>) -> StreamExecutorResult<()>
```

Comment:
I think the comments are verbose (and a little outdated?). 

Function name `set_states` might be enough to describe what it does.